src/lib.rs

```diff
@@ -256,8 +285,93 @@ impl Navigator {
+    ///
+    /// `prescale_value = round(clock_freq / (4096 * desired_freq)) - 1`.
+    ///
+    /// The minimum prescaler value is 3, which corresonds to 1526 Hz.
+    /// The maximum prescaler value is 255, which corresponds to 24 Hz.
+    ///
+    /// If you want to control a servo, set a prescaler value of 100. This will
+    /// correspond to a frequency of about 60 Hz, which is the frequency at
+    /// which servos work.
+    ///
+    /// Internally this function stops the oscillator and restarts it after
+    /// setting the prescaler value if it was running.
+    ///
+    /// Re-run the set_pwm_channel_value() is required.
+    ///
+    /// # Further info
+    /// Check **[7.3.5 - PWM frequency PRE_SCALE](https://www.nxp.com/docs/en/data-sheet/PCA9685.pdf#page=25)**
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use navigator_rs::{pwm_Channel, Navigator};
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    /// nav.pwm_enable();
+    /// nav.set_pwm_freq_prescale(100); // sets the pwm frequency to 60 Hz
+    /// nav.set_pwm_channel_value(pwm_Channel::C0, 2048); //sets the duty cycle to 50%
+    ///
+    /// ```
+    pub fn set_pwm_freq_prescale(&mut self, mut value: u8) {
+        if value < 3 {
+            warn!("Invalid value. Value must be greater than 3.");
+            value = 3
+        }
+        self.pwm.set_prescale(value).unwrap();
+
+        let clamped_freq = NAVIGATOR_PWM_XTAL_CLOCK_FREQ / (4_096.0 * (value as f32 + 1.0));
+        info!(
+            "PWM frequency set to {:.2} Hz. Prescaler value: {}",
+            clamped_freq, value
+        );
+    }
+
+    /// Sets the pwm frequency in Hertz of [`Navigator`].
+    ///
+    /// The navigator module uses a crystal with 24.5760 MHz clock. You can set a value for a frequency beetwen 24 and 1526 Hz.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use navigator_rs::{pwm_Channel, Navigator, SensorData};
+    /// use std::thread::sleep;
+    /// use std::time::Duration;
+    ///
+    /// let mut nav = Navigator::new();
+    /// nav.init();
+    /// nav.pwm_enable();
+    /// let mut i:f32 = 10.0;
+    ///
+    /// loop {
+    ///     nav.set_pwm_freq_hz(i); // sets the pwm frequency to 60 Hz
+    ///     nav.set_pwm_channel_value(pwm_Channel::C0, 2048); //sets the duty cycle to 50%
+    ///     i=i+10.0;
+    ///     sleep(Duration::from_millis(1000));
+    /// }
+    /// ```
+    pub fn set_pwm_freq_hz(&mut self, mut freq: f32) {
+        if freq < 24.0 {
+            warn!("Invalid value. Value must be greater than 24.0.");
+            freq = 24.0;
+        }
+
+        if freq > 1526.0 {
+            warn!("Invalid value. Value must be less than 1526.0.");
+            freq = 1526.0;
+        }
+
+        let prescale_clamped_value =
+            (NAVIGATOR_PWM_XTAL_CLOCK_FREQ / (4_096.0 * freq)).round() as u8 - 1;
+
+        self.set_pwm_freq_prescale(prescale_clamped_value);
     }
```

Comment:
```suggestion
    pub fn set_pwm_freq_hz(&mut self, mut freq: f32) {
        let min_freq = 24.0;
        let max_freq = 1526.0;
        if freq < min_freq {
            warn!("Invalid value. Value must be greater than {min_freq}.");
            freq = min_freq;
        }

        if freq > max_freq {
            warn!("Invalid value. Value must be less than {max_freq}.");
            freq = max_freq;
        }

        let prescale_clamped_value =
            (NAVIGATOR_PWM_XTAL_CLOCK_FREQ / (4_096.0 * freq)).round() as u8 - 1;

        self.set_pwm_freq_prescale(prescale_clamped_value);
    }
```

Reply:
should be `greater/less than or equal to`, not just `greater/less than`.